src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,83 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo }
         };
+  }
+
+  handleSettingsClick = () => {
+    let indexOf = -1;
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+
+ 
+      indexOf = this.props.svgOptions.findIndex(this.props.value);
+
+    if (this.state.IndexOf < 0) {
+      this.props.svgOptions.push(this.props.value);
+    } else {
+      this.props.svgOptions.splice(this.state.IndexOf, 1);
+        indexOf = -1;
```

Comment:
isnt indexOf already -1 at this point, did you read the documentation https://www.w3schools.com/jsref/jsref_findindex.asp

Reply:
and even if you set it, this value isnt going to be used in this function anymore

Reply:
yhea surely, I just did pay attention when changing.

Reply:
I have removed.